app/src/main/java/com/android/sample/model/user/UserRepositoryFirestore.kt

```diff
@@ -0,0 +1,86 @@
+package com.android.sample.model.user
+
+import android.util.Log
+import com.google.android.gms.tasks.Task
+import com.google.firebase.Firebase
+import com.google.firebase.auth.auth
+import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.FirebaseFirestore
+
+class UserRepositoryFirestore(private val db: FirebaseFirestore) : UserRepository {
+  private val collectionPath = "users"
+
+  override fun init(onSuccess: () -> Unit) {
+    if (Firebase.auth.currentUser != null) {
+      onSuccess()
+    }
```

Comment:
Thank you for the suggestion but I don't think it should be placed in UserRepositoryFirestore.kt file because this file only serves as a way of communication between the Firestore database and the app. So, it is only for when the user is online, and not offline.
In the future, we will implement the offline mode and at that time, we will add the cache.